runtime/callback.c

```diff
@@ -447,6 +447,7 @@ static unsigned int hash_value_name(char const *name)
 
 CAMLprim value caml_register_named_value(value vname, value val)
 {
+  CAMLparam2(vname, val);
   const char * name = String_val(vname);
```

Comment:
I suggest adding a comment about `name` as this seems like a footgun (unsafe to use after line 456).

Reply:
commented (and set to NULL for good measure).